controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -236,6 +236,9 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 			}
 			if instance.Spec.Monitoring.ManagementState == operatorv1.Managed {
 				log.Info("Monitoring enabled, won't apply changes", "cluster", "Self-Managed RHODS Mode")
+				if err := r.configureSegmentIO(ctx, instance); err != nil {
+					return reconcile.Result{}, err
+				}
```

Comment:
I imagine this is all in the function `configureSegmentIO` -- but from your description:

> * if DSCI monitoring is Managed

Not sure we need to tie this to `Managed` monitoring, right? Since it's not using the monitoring stack?

> * if user has not disable tracking data usage in Dashboard

Do you check our OdhDashboardConfig? 

> * if odhdashbaordconfig has disableTracking:false
> then some track data will be sent to internal system.

This should be on us to find the configmap and make use of it -- not sure the Operator needs to do this logic too...

Reply:
i did not want to change the logic from old 2.16.0 if it is just to get it back without improvement.
it was done in such a way, that user could set monitoring to Removed in DSCI and disable the function.

the checks on 
- if user has not disable tracking data usage in Dashboard
- if odhdashbaordconfig has disableTracking:false

are not done by Operator, 
the only thing in the PR is Operator create CM if DSCI monitoring is Managed

I put all three conditions in the PR purely for QE to understand the logic if want to test it.